File: ecs-java-s3-workshop/src/main/java/com/emc/ecs/s3/sample/_04_DeleteObject.java
/*
 * Copyright 2013 EMC Corporation. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.txt
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
package com.emc.ecs.s3.sample;

import com.emc.object.s3.S3Client;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class _04_DeleteObject {

	public static void main(String[] args) throws Exception {
    	// create the ECS S3 Client
    	S3Client s3 = ECSS3Factory.getS3Client();

    	// retrieve the object value from user
        System.out.println( "Enter the object key:" );
        String key = new BufferedReader( new InputStreamReader( System.in ) ).readLine();
        
        // delete the object from the demo bucket
        s3.deleteObject(ECSS3Factory.S3_BUCKET, key);
        
        // print out key/value for validation
    	System.out.println( String.format("object [%s/%s] deleted",
    			ECSS3Factory.S3_BUCKET, key));
    }
}

File: ecs-net-workshop/aws-net-workshop/examples/_04_DeleteObject.cs
using System;
using ECSSDK.S3;
using Amazon.S3.Model;

namespace ecs_net_workshop.examples
{
    class _04_DeleteObject
    {
        public static void Main(string[] args)
        {
            // create the ECS S3 client
            ECSS3Client s3 = ECSS3Factory.getS3Client();

            // retrieve the object key from user
            Console.Write("Enter the object key: ");
            string key = Console.ReadLine();

            // create the request object
            DeleteObjectRequest request = new DeleteObjectRequest()
            {
                BucketName = ECSS3Factory.S3_BUCKET,
                Key = key
            };

            // delete the object in the demo bucket
            s3.DeleteObject(request);


            // print out object key/value for validation
            Console.WriteLine(string.Format("Object {0}/{1} deleted", ECSS3Factory.S3_BUCKET, key));
            Console.ReadLine();
        }
    }
}
